### PR TITLE
Support T5 benchmarking in transformers tool

### PR DESCRIPTION
**Pull request by wangyems**
**Description**: Describe your changes.

**Motivation and Context**
- Why is this change required? What problem does it solve?
- If it fixes an open issue, please link to the issue here.

### CODE REVIEW

File: onnxruntime/python/tools/transformers/benchmark.py [hunk 1]
```diff
@@ -286,9 +299,17 @@ def run_tensorflow(use_gpu, model_names, model_class, precision, batch_sizes, se
                 input_ids = tf.constant(values, shape=(batch_size, sequence_length), dtype=tf.int32)
 
                 try:
-                    model(input_ids, training=False)
+                    def encoder_forward():
+                        return model(input_ids, training=False)
+
+                    def encoder_decoder_forward():
+                        return model(input_ids, decoder_input_ids=input_ids, training=False)
```
Comment (tianleiwu) [inline onnxruntime/python/tools/transformers/benchmark.py:306]: I think transformers has changed to make decoder_input_ids optional:
https://github.com/huggingface/transformers/commit/6912265711fdf29e537e35194552bfa2c31b72bc
Please install transformers from source and try it.

File: onnxruntime/python/tools/transformers/huggingface_models.py [hunk 2]
```diff
@@ -89,11 +89,11 @@
...
-    #"t5-small": (["input_ids"], 11, False, "bert"),
-    #"t5-base": (["input_ids"], 11, False, "bert"),
-    #"t5-large": (["input_ids"], 11, False, "bert"),
-    #"t5-3b": (["input_ids"], 11, False, "bert"),
-    #"t5-11b": (["input_ids"], 11, False, "bert"),
+    "t5-small": (["input_ids"], 11, False, "bert"),
+    "t5-base": (["input_ids"], 11, False, "bert"),
+    "t5-large": (["input_ids"], 11, False, "bert"),
+    "t5-3b": (["input_ids"], 11, False, "bert"),
+    "t5-11b": (["input_ids"], 11, False, "bert"),
```
Comment (yufenglee) [inline onnxruntime/python/tools/transformers/huggingface_models.py:96]: It is really good that the tool can benchmark almost all the transformer based models

File: onnxruntime/python/tools/transformers/onnx_exporter.py [hunk 3]
```diff
@@ -223,7 +223,7 @@ def load_pretrained_model(model_name, config, cache_dir, custom_model_class, if_
     transformers_module = __import__("transformers", fromlist=[model_class_name])
     model_class = getattr(transformers_module, model_class_name)
 
-    return model_class.from_pretrained(model_name, config=config, cache_dir=cache_dir)
+    return model_class.from_pretrained(model_name, config=config, cache_dir=cache_dir, use_cdn=False)
```
Comment (yufenglee) [inline onnxruntime/python/tools/transformers/onnx_exporter.py:226]: why set use_cdn to False for all models? Its spec says only model > 20g needs to set it to False.

File: onnxruntime/python/tools/transformers/benchmark.py (hunk 1 above)
Comment (wangyems) [reply onnxruntime/python/tools/transformers/benchmark.py:306]: it looks like current master of transformers can work without current change for torchscript benchmark

File: onnxruntime/python/tools/transformers/onnx_exporter.py (hunk 3 above)
Comment (wangyems) [reply onnxruntime/python/tools/transformers/onnx_exporter.py:226]: only t5-11b requires that actually. I think it's ok to set it to false for all models(may lower download speed according to spec). But It's better to only set for t5-11b. Will modify it.

File: onnxruntime/python/tools/transformers/benchmark.py (hunk 1 above)
Comment (tianleiwu) [reply onnxruntime/python/tools/transformers/benchmark.py:306]: I suggest we wait for next release of transformers package. They usually have short release interval (2~4 weeks).

Comment (wangyems) [reply onnxruntime/python/tools/transformers/benchmark.py:306]: ok

**Review by tianleiwu — APPROVED** [review]

**Pull request merged**